src/main/java/dev/socketmods/socketnukes/explosion/types/VanillaExplosionType.java

```diff
@@ -0,0 +1,244 @@
+                            double currentEntityDistanceZ = currentEntity.getPosZ() - source.getZ();
+                            double pythagoreanDistance = MathHelper.sqrt(currentEntityDistanceX * currentEntityDistanceX + currentEntityDistanceY * currentEntityDistanceY + currentEntityDistanceZ * currentEntityDistanceZ);
+                            if (pythagoreanDistance != 0.0D) {
+                                currentEntityDistanceX = currentEntityDistanceX / pythagoreanDistance;
+                                currentEntityDistanceY = currentEntityDistanceY / pythagoreanDistance;
+                                currentEntityDistanceZ = currentEntityDistanceZ / pythagoreanDistance;
+                                double rayLength = Explosion.getBlockDensity(explosionPos, currentEntity);
+                                double damageFalloff = (1.0D - currentEntityDistanceToExplosion) * rayLength;
+                                currentEntity.attackEntityFrom(this.getDamageSource(), (float) ((int) ((damageFalloff * damageFalloff + damageFalloff) / 2.0D * 7.0D * (double) radiusx2 + 1.0D)));
+                                if (currentEntity instanceof LivingEntity) {
+                                    damageFalloff = ProtectionEnchantment.getBlastDamageReduction((LivingEntity) currentEntity, damageFalloff);
+                                }
+
+                                currentEntity.setMotion(currentEntity.getMotion().add(currentEntityDistanceX * damageFalloff, currentEntityDistanceY * damageFalloff, currentEntityDistanceZ * damageFalloff));
+                                if (currentEntity instanceof PlayerEntity) {
+                                    PlayerEntity playerentity = (PlayerEntity) currentEntity;
+                                    if (!playerentity.isSpectator() && (!playerentity.isCreative() || !playerentity.abilities.isFlying)) {
+                                        meta.entityDisplacements.put(playerentity, new Vector3d(currentEntityDistanceX * damageFalloff, currentEntityDistanceY * damageFalloff, currentEntityDistanceZ * damageFalloff));
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+                return meta;
+            case STAGE_BLOCKS:
+                if (worldIn.isRemote) {
+                    worldIn.playSound(source.getX(), source.getY(), source.getZ(), properties.getExplosionSound(), SoundCategory.BLOCKS, 4.0F, (1.0F + (worldIn.rand.nextFloat() - worldIn.rand.nextFloat()) * 0.2F) * 0.7F, false);
+                }
+
+                if (properties.doesMakeParticles()) {
+                    worldIn.addParticle(properties.getParticleToEmit(), source.getX(), source.getY(), source.getZ(), 1.0D, 0.0D, 0.0D);
+                }
+
+                if (doBlocksDrop) {
```

Comment:
This is also used in the `CubicExplosionType`

Reply:
Ditto